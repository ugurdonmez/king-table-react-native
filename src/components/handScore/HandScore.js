import React from 'react';
import { View, Text, StyleSheet } from 'react-native'

const handScore = (props) => {
    return (
        <Text style={styles.container}>
            {props.score}
        </Text>
    );
};

export default handScore;

var styles = StyleSheet.create({
    container: {
        textAlign: 'center',
        borderRadius: 4,
        borderWidth: 0.5,
        borderColor: '#d6d7da',
        width: '20%',
    },
})
