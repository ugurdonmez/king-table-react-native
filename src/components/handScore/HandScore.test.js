import React from 'react';
import 'react-native';
import { shallow } from 'enzyme';
import HandScore from './HandScore';
import renderer from 'react-test-renderer'

describe('<HandScore />', () => {
    it('renders without crashing', () => {
        shallow(<HandScore />)
    })

    it('match snapshot', () => {
        const tree = renderer.create(
            <HandScore
                score="10"
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    })

    it('text should be equal to prop name', () => {
        const wrapper = renderer.create(
            <HandScore 
                name="10"
            />
        ).toJSON();

        expect(wrapper.children[0]).toBe('10');
    })
})