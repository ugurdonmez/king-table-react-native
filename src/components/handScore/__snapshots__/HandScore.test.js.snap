// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HandScore /> match snapshot 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  style={
    Object {
      "borderColor": "#d6d7da",
      "borderRadius": 4,
      "borderWidth": 0.5,
      "textAlign": "center",
      "width": "20%",
    }
  }
>
  100
</Text>
`;
